py/types.py

```diff
@@ -0,0 +1,16 @@
+from collections.abc import Sequence, Mapping
+from typing import TypedDict, Protocol
+
+
+class AIProvider(Protocol):
+
+  def __init__(self, config: Mapping[str, str|float|dict]) -> None:
+      pass
+
+  def request(self, messages: Sequence[Message]) -> List[str]:
+      pass
+
+class Message(TypedDict):
+    role: str
+    content: str
```

Comment:
This would need to be changed to the new structured message. Now also support image_url type. E.g. see [test_parse_include_image_message](https://github.com/madox2/vim-ai/blob/main/tests/chat_test.py#L256C5-L256C37)